### PR TITLE
Refactor documentation and error messages

### PR DESCRIPTION
**Pull request by whatisinternet**
#### Thanks for the great library! :+1:

**What this PR does:**
1. Clarifies some of the error messages wording. 
2. Fixes some clarity and spelling issues on the README.
3. Updates the test to work with the new wording.
4. Bumps the version to 0.1.2

**Why it does it:**
1. Some of the error messages lacked complete meaning. This caused confusion as to why exactly tests were failing. 
2. The readme while great, needed some minor changes.
3. Some of the tests depend on the exact text of the error message. The updated tests provide the needed changes.
4. This is not a breaking change thus only a point release.

### CODE REVIEW

File: src/toHaveSameItems.js [hunk 1]
```diff
@@ -70,17 +70,17 @@ beforeEach(function() {
             return {
                 compare: function(actual, expected, ignoreOrder) {
                     if(!Array.isArray(actual) && !isObject(actual)) {
-                        throw new Error('Actual must be an Array or Object');
+                        throw new Error('Actual must be an Array or Object. Is type: ' + typeof(actual));
```
Comment (just-boris) [inline src/toHaveSameItems.js]: `typeof` is an operator, not a function. So, you don't need to use parenthesis here.

File: src/toHaveSameItems.js [hunk 2]
```diff
@@ -70,17 +70,17 @@ beforeEach(function() {
...
                     }
                     if(!Array.isArray(expected) && !isObject(expected)) {
-                        throw new Error('Expectation must be an Array or Object');
+                        throw new Error('Expectation must be an Array or Object. Is type: ' + typeof(expected));
```
Comment (just-boris) [inline src/toHaveSameItems.js]: Same here

File: src/toHaveUniqueItems.js [hunk 3]
```diff
@@ -11,17 +11,17 @@ beforeEach(function() {
...
-                return pass ? 'All items in array are unique' : 'Array contains duplicates: \n'+duplicates.map(function(dupe) {
+                return pass ? 'All items in the array are unique' : 'Array contains duplicates: \n'+duplicates.map(function(dupe) {
                     return JSON.stringify(dupe[2])+' at '+dupe[0]+' and '+dupe[1];
                 }).join('\n');
             }
             return {
                 compare: function(actual, expected) {
                     if(!Array.isArray(actual)) {
-                        throw new Error("Actual is not Array");
+                        throw new Error("Actual is not Array. It is of type: " + typeof(actual));
```
Comment (just-boris) [inline src/toHaveUniqueItems.js]: And here

File: test/toHaveUniqueItems.spec.js [hunk 4]
```diff
@@ -4,13 +4,13 @@ describe("toHaveUniqueItems", function() {
     it("should accept Array in actual", function() {
         expect(function() {
             expect("test").not.toHaveUniqueItems();
-        }).toThrowError("Actual is not Array")
+        }).toThrowError("Actual is not Array. It is of type: " + typeof("test"))
```
Comment (just-boris) [inline test/toHaveUniqueItems.spec.js]: Why not just

```
"Actual is not Array. It is of type: string"
```

This seems to me easier to read.

**Comment by just-boris** [comment]
Thanks for your help. It was so helpful for me.
Please, fix some issues with `typeof` and then I can merge this

**Comment by whatisinternet** [comment]
No problem! :smile: 
Issues are fixed, tests pass.

Thanks again!

File: test/toHaveUniqueItems.spec.js (hunk 4 above)
Comment (whatisinternet) [inline test/toHaveUniqueItems.spec.js]: I missed that this comment was in the test. I guess that makes more sense. I'll change in and push up again.

**Comment by whatisinternet** [comment]
Should be good to go now. Let me know if you see anything else.

**Comment by just-boris** [comment]
Merged, thanks!

**Pull request merged**